harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -263,85 +578,278 @@ def cleanup_iterator() -> None:
                 for optimizer in self.context.optimizers:
                     hvd.broadcast_optimizer_state(optimizer, root_rank=0)
 
-            with self.prof:
-                for callback in self.callbacks.values():
-                    with self.prof.record_timing(
-                        f"callbacks.{callback.__class__.__name__}.on_training_start"
-                    ):
-                        callback.on_training_start()
-                self._run()
+            exit_stack.enter_context(self.prof)
+            for callback in self.callbacks.values():
+                with self.prof.record_timing(
+                    f"callbacks.{callback.__class__.__name__}.on_training_start"
+                ):
+                    callback.on_training_start()
+
+            self._run()
+
+    @staticmethod
+    def dataloader_next(
```

Comment:
no reason for this to be a static function, right?  Make it a standalone function at the top of the file?